build_tools/bazel/build.sh

```diff
@@ -21,13 +21,23 @@ set -e
 set -x
 
 # CI-friendly defaults that control availability of certain platform tests.
-if ! [[ -v IREE_VULKAN_DISABLE ]]; then
-  IREE_VULKAN_DISABLE=1
-fi
-test_env_args=(
+IREE_VULKAN_DISABLE="${IREE_VULKAN_DISABLE:-1}"
+declare -a test_env_args=(
   --test_env=IREE_VULKAN_DISABLE=$IREE_VULKAN_DISABLE
 )
-echo "Running with test env args: ${test_env_args[@]}"
+
+declare -a default_build_tag_filters=("-nokokoro")
+declare -a default_test_tag_filters=("-nokokoro")
+
+# We can still build things that use vulkan. Only add to test tag filters.
+if [[ "${IREE_VULKAN_DISABLE?}" == 1 ]]; then
+  default_test_tag_filters+=("-driver=vulkan")
+fi
+BUILD_TAG_FILTERS=("${BUILD_TAG_FILTERS:-${default_build_tag_filters[@]?}}")
+TEST_TAG_FILTERS=("${TEST_TAG_FILTERS:-${default_test_tag_filters[@]?}}")
+
+BUILD_TAG_FILTER_STRING="$(IFS="," ; echo "${BUILD_TAG_FILTERS[*]?}")"
+TEST_TAG_FILTER_STRING="$(IFS="," ; echo "${TEST_TAG_FILTERS[*]?}")"
```

Comment:
My bash is rusty - what are these symbols (`[@]`, `[*]`, `IFS=","`) doing? This is "cleaner" than tags :P?

Reply:
That pattern is basically "join on commas". I can extract a function if you'd like. I think we should also consider rewriting this whole thing in python... Agreed that this is not super readable.

The bit about it being cleaner was about the bazel query itself. This PR makes everything configurable, which is why it adds complexity.

Reply:
No need to extract to a function. I usually like to comment on intent and/or provide samples of inputs/outputs when using convoluted syntax (e.g. regex, string manipulation). Since this is a generic build script (primarily used by the CI), this could use more explanation around what these lines are doing and how developers (or other scripts) invoking this script interface with this logic.

Reply:
Ok simplified some of the array nonsense and documented things better. Thanks for keeping me honest :-)

Reply:
Oh looks like you already saw and re-approved :-) Now has more documentation.